### PR TITLE
Cold-leaching and smoking acorns

### PR DESCRIPTION
**Pull request by Sathra225**
#### Summary
Content "Mass prepping and cooking acorns"

#### Purpose of change
Acorns are the last vanilla nut without a mass-prep method, fixes that. Also allows cooked acorns to be smoked into roasted and eaten frozen because it is edible frozen, just not great.

#### Describe the solution
Two new recipe lines (tank/pot of blanching acorns) that involve hulling a pile of acorns and putting them into either a large clay pot (for innawoods players) or 60L tank (everyone). This item goes through a 6-day wait (same as juve sourdough) and turns into a tank/pot of 'blanched acorns' via delayed transform. This is then disassembled into the pot/tank and blanched acorns, which can then be smoked or cooked into 'cooked acorns'. Which can then be eaten, cooked, milled or smoked again into 'roasted acorns'.

#### Describe alternatives you've considered
Fermenting maybe? Some kind of method for leaving them in running water for a week?
Multistage leaching was also an option, but also insane.

#### Testing
Made changes, loaded game. Spawned in all components, made the blanching items. Debug-skipped time till they were ready (and confirmed not-ready message was working). Disassembled both blanchers, ate some, cooked some, smoked some, smoked some again.

#### Additional context
Got the basic process from here.
https://www.tyrantfarms.com/diy-how-to-make-acorn-flour-acorn-recipes/
Weights of blanchers are based on weight of all components, rounded down to account for not including the shells. Adding a second delayed transform for sun-drying would just be excessive I think.

### CODE REVIEW

File: data/json/recipes/recipe_deconstruction.json [hunk 1]
```diff
@@ -701,38 +701,6 @@
...
-    "components": [ [ [ "fur", 1 ] ] ]
-  },
-  {
-    "result": "bikini_top_leather",
-    "type": "uncraft",
-    "time": "20 s",
-    "activity_level": "NO_EXERCISE",
-    "qualities": [ { "id": "CUT_FINE", "level": 1 } ],
-    "components": [ [ [ "leather", 1 ] ] ]
-  },
```
Comment (snipercup) [inline data/json/recipes/recipe_deconstruction.json:735]: Pretty sure you don't need to delete these uncraft recipes for this PR

Comment (Sathra225) [reply data/json/recipes/recipe_deconstruction.json:735]: Oh damn, thanks for the find. Fixing now

File: data/json/items/tool/cooking.json [hunk 2]
```diff
@@ -1051,5 +1051,89 @@
     "pocket_data": [ { "pocket_type": "MAGAZINE_WELL", "flag_restriction": [ "GAS_TANK" ], "default_magazine": "small_propane_tank" } ],
     "charges_per_use": 1,
     "use_action": [ "HOTPLATE", "HEAT_FOOD" ]
+  },
+  {
+    "id": "blanching_acorns_clay",
+    "type": "GENERIC",
+    "category": "food",
+    "name": { "str_sp": "pot of blanching acorns" },
```
Comment (Night-Pryanik) [inline data/json/items/tool/cooking.json:1059]:
```suggestion
    "name": { "str": "pot of blanching acorns", "str_pl": "pots of blanching acorns" },
```

Comment (Sathra225) [reply data/json/items/tool/cooking.json]: thanks, updating all of them. Couldn't remember what the strings were.

**Comment by Qrox** [comment]
>  This item goes through a 6-day wait (same as juve sourdough) and turns into a tank/pot of 'blanched acorns' via delayed transform. This is then disassembled into the pot/tank and blanched acorns

Sourdough no longer requires specific containers after #56324, maybe the same can be done to blanched acorns?

**Comment by Sathra225** [comment]
Its not entirely similar, as this is abstracting a week of water changes and stirring. Its not a small amount either, as the quantities in the recipes are close to the capacity of the containers themselves.

**Comment by Sathra225** [comment]
I did try out having them as a transforming liquid and it works but with one major problem.
The blanched acorns are rotten at the end. Doing it through containers and disassembling circumvents this.
(I also discovered that I'd forgotten to _include water in the recipes_)

**Comment by Qrox** [comment]
Are blanched acorns rotten because the transformation time is longer than the spoiling time? If that's the case perhaps setting the spoiling time to 11 (6+5) days might prevent it from rotting?

**Comment by Sathra225** [comment]
Yeah, that only fixes it if you constantly check. Have to remember that this is actually an abstraction of the IRL process, which involves stirring and changing the water once or twice a day.
As I understand it, a week is just the average time, it can take longer (or less depending on species). It shouldn't spoil if cared for properly due to the tannins and water changes. So having them start spoiling only _after_  decanting works better as the acorns are more exposed to airborne spores and whatnot while still being water-heavy.

**Comment by Qrox** [comment]
> having them start spoiling only after decanting

I think that can be done by implementing a "reset_birthday" flag in the transformation entry, but it will require c++ change.

**Comment by Sathra225** [comment]
Which I can't do because C++ is dark magic to me.

**Comment by dseguin** [comment]
You could implement a `reset_birthday` field like this:

```diff
diff --git a/src/iuse_actor.cpp b/src/iuse_actor.cpp
index 0156a75fc4..eddd9adb18 100644
--- a/src/iuse_actor.cpp
+++ b/src/iuse_actor.cpp
@@ -207,6 +207,8 @@ void iuse_transform::load( const JsonObject &obj )
     obj.read( "qualities_needed", qualities_needed );
 
     obj.read( "menu_text", menu_text );
+
+    obj.read( "reset_birthday", reset_birthday );
 }
 
 cata::optional<int> iuse_transform::use( Character &p, item &it, bool t, const tripoint &pos ) const
@@ -328,6 +330,9 @@ cata::optional<int> iuse_transform::use( Character &p, item &it, bool t, const t
     }
     obj->item_counter = countdown > 0 ? countdown : obj->type->countdown_interval;
     obj->active = active || obj->item_counter;
+    if( reset_birthday ) {
+        obj->set_birthday( calendar::turn );
+    }
     if( p.is_worn( *obj ) ) {
         if( !obj->is_armor() ) {
             item_location il = item_location( p, obj );
diff --git a/src/iuse_actor.h b/src/iuse_actor.h
index 10d80694c6..a69e8c24ba 100644
--- a/src/iuse_actor.h
+++ b/src/iuse_actor.h
@@ -80,6 +80,9 @@ class iuse_transform : public iuse_actor
         /**does the item requires to be wielded to be activable*/
         bool need_wielding = false;
 
+        /** Whether to reset this item's age on transformation*/
+        bool reset_birthday = false;
+
         /** subtracted from @ref Creature::moves when transformation is successful */
         int moves = 0;
```

Then you'd just add `"reset_birthday": true` to your transform action.

**Comment by Sathra225** [comment]
I...will have to take your word on that, as I have no idea what I'm looking at.
git command line file modification?

**Comment by Qrox** [comment]
Actually I tested with sourdough starter by adding spoil time to it and it did not spoil after transformation. In fact it did not spoil long after the specified spoil time so it might be a bug. (Also it seems the rot timer need to be reset if it did spoil, not the item birthday.)

> I did try out having them as a transforming liquid and it works, but with one major problem.
The blanched acorns are rotten at the end.

Could you post your code so I can test with it?

**Comment by Sathra225** [comment]
In recipes/food/vegetable_dishes.json:
  {
    "type": "recipe",
    "result": "blanching_acorns",
    "activity_level": "MODERATE_EXERCISE",
    "category": "CC_FOOD",
    "subcategory": "CSC_FOOD_VEGGI",
    "skill_used": "cooking",
    "skills_required": [ "survival", 2 ],
    "difficulty": 3,
    "time": "2 m",
    "batch_time_factors": [ 20, 1 ],
    "autolearn": true,
    "qualities": [ { "id": "HAMMER", "level": 2 } ],
    "proficiencies": [ { "proficiency": "prof_food_prep" }, { "proficiency": "prof_forage_cooking" } ],
    "tools": [ [ [ "bag_canvas", -1 ], [ "pillowcase", -1 ], [ "sheet", -1 ] ] ],
    "components": [ [ [ "acorns", 4 ] ], [ [ "water", 8 ], [ "water_clean", 8 ] ] ]
  }
And in items/comestibles/nuts.json:
  {
    "type": "GENERIC",
    "id": "blanching_acorns",
    "name": { "str_sp": "blanching acorns" },
    "copy-from": "acorns",
    "description": "Water and hulled acorns, slowly leaching the tannins.  It'll take about a week.",
    "price": 90,
    "price_postapoc": 200,
    "phase": "liquid",
    "delete": { "flags": [ "EDIBLE_FROZEN" ] },
    "use_action": {
      "target": "acorns_blanched",
      "msg": "After a week of changing water and taste-testing, the acorns are no longer bitter and ready to decant.",
      "moves": 50,
      "type": "delayed_transform",
      "transform_age": 518400,
      "not_ready_msg": "The acorns are still bitter."
    }
  }

**Comment by Qrox** [comment]
I created a PR #57824 to address the rotting problem. It now keeps the relative rot instead of absolute rot during item transformation.

> It's also somewhat annoying from a player perspective as you need to have the container in your inventory to (a)ctivate the liquid to transform.

This is also an issue with fermenting yogurt and presumably other liquid items with delayed transformation, so it is a separate issue to fix, and I still consider separating the container and transformed item is the correct path as it allows any container to be used.

**Comment by Sathra225** [comment]
Fine, whatever. Closing. I'll make a new one after yours is merged.

**Pull request closed without merging**